packages/appstore/src/components/modals/wallets-upgrade-modal/wallets-upgrade-modal.tsx

```diff
@@ -1,64 +1,72 @@
 import React from 'react';
 import { useWalletMigration } from '@deriv/hooks';
-import { Button, Icon, Text, Modal } from '@deriv/components';
+import { Button, Text, Modal } from '@deriv/components';
 import { observer, useStore } from '@deriv/stores';
 import { localize, Localize } from '@deriv/translations';
 import './wallets-upgrade-modal.scss';
-
-const MODAL_TRANSITION_TIMEOUT_MS = 250; // matching the default one
+import classNames from 'classnames';
+import { getUrlBase } from '@deriv/shared';
 
 const WalletsUpgradeModal = observer(() => {
     const { traders_hub, ui } = useStore();
-    const { toggleWalletsUpgrade } = traders_hub;
+    const { is_real_wallets_upgrade_on, toggleWalletsUpgrade } = traders_hub;
     const { is_mobile, is_desktop } = ui;
-    const { is_eligible } = useWalletMigration();
+    const { is_eligible, startMigration } = useWalletMigration();
     const isWalletMigrationModalClosed = localStorage.getItem('is_wallet_migration_modal_closed');
     const [modalOpen, setModalOpen] = React.useState(!isWalletMigrationModalClosed);
 
     const closeModal = () => {
         setModalOpen(false);
         localStorage.setItem('is_wallet_migration_modal_closed', 'true');
+        toggleWalletsUpgrade(false);
+    };
+
+    const handleMigration = () => {
+        closeModal();
+        startMigration();
     };
 
     return (
         <Modal
             className='wallets-upgrade-modal'
-            is_open={is_eligible && modalOpen}
-            width='60rem'
+            is_open={(is_eligible && modalOpen) || is_real_wallets_upgrade_on}
+            width={is_mobile ? '32.8rem' : '116rem'}
             title=' '
             toggleModal={closeModal}
-            transition_timeout={MODAL_TRANSITION_TIMEOUT_MS}
         >
             <Modal.Body>
                 <div className='wallets-upgrade-modal__content'>
-                    <Icon
-                        icon={`IcAppstoreWalletsUpgradeCoins${is_mobile ? 'Horizontal' : ''}`}
-                        width={is_mobile ? 190 : 300}
-                        height={is_mobile ? 80 : 300}
-                        className='wallets-upgrade-modal__image'
-                        data_testid={`dt_wallets_upgrade_coins${is_mobile ? '_horizontal' : ''}`}
-                    />
-                    <div className='wallets-upgrade-modal__description'>
-                        <Text align={is_mobile ? 'center' : 'left'} size={is_mobile ? 's' : 'm'} weight='bold'>
+                    <div className='wallets-upgrade-modal__media-container'>
+                        <video
```

Comment:
@aizad can we embed the video in an iframe instead?

Reply:
@nijil-deriv our best bet is to use cloudflare to stream it.

Reply:
Seems like the option might be too late because this is an urgent card. I'll make the changes in the next sprint but for now in my opinion this works fine.